### PR TITLE
[Bugfix] Fix export routing for quiz exercises

### LINKED ISSUE #3317: Breadcrumbs not displayed correctly for Export Quiz
### Describe the bug

When exporting a quiz the breadcrumbs are not displayed properly.

#### To Reproduce

1. Create a quiz
2. Open the exercise overview
3. Click on _Export Quiz Exercises_
4. The breadcrumb on the upper left side of the screen is not displayed properly

#### Expected behavior

The breadcrumb should display the course name instead of `{{ param }}` 

#### Screenshots

Not properly displayed breadcrumb
image

Execution of steps to produce the bug
ExportQuizBreadcrumbIssue

### Environment

<details><pre>
 - OS: Windows 10
 - Browser: latest Chrome
 - Artemis version: 4.11.8 
</pre></details>

### LINKED ISSUE #3433: Export Quiz Exercise forwards to the Create Quiz page
### Describe the bug

The _Export Quiz Exercise_ button forwards the user to the _Create Quiz_ page. Also the breadcrumb is not displayed properly.

#### To Reproduce

1. Log in as Editor/Instructor
2. Click on _Export Quiz Exercises_ (the course needs to contain at least one quiz)
3. See that you are forwarded to the _Create Quiz_ page

#### Expected behavior

_Export Quiz Exercises_ should be an option to export quizzes as json files. The button should probably lead to a page where quizzes / quiz questions that shall be exported can be selected.

#### Screenshots

Button that should let the user export quizzes as json file:
image (2)

Page the user is forwarded to:
image (3)

### Environment

<details><pre>
 - Artemis Version: 4.12.2
</pre></details>

### PR DESCRIPTION
**Pull request by imstreetspirit**
### Motivation and Context

Fixes #3433 
Fixes #3317

- Problem is that export button does not work for quiz exercises. Export navigates to Create New Quiz Page. Routing definitions were correct, but I changed the order of the routes in the `quizManagementRoute`. (just move up the 'new' and 'export' routes) According to [official documentation](https://angular.io/guide/router#route-order), the router uses a first-match wins strategy when matching routes, so more specific routes should be placed above less specific routes. After reordering, routing works again. 

- When the user opened Export Quiz Exercise page, the last segment was wrong in the breadcrumb. Export string is set with wrong translation. I also fixed this in the `NavbarComponent`

- In addition, I re-organized the authorizations in quiz management routes.

### Steps for Testing

Following steps cover only export routing. **Please check that all routes  related with quiz exercises are working properly in course and exam. e.g Preview, Solutions, Create, Edit, Re-Evaluate, Submissions, Delete.**

1. Log in to Artemis as INSTRUCTOR
2. Navigate to Course Management
3. Go to Exercises for the course, to see Export Quiz Exercises button, you need to have at least one quiz exercise.
4. click that, it should open the correct page, not the Create New Quiz Exercise page.
5. look at the breadcrumbs in that page, it should be correct (see screenshot)

### Test Coverage

nothings changed

### CODE REVIEW

**Review by ndricimrr — APPROVED** [review]
Tested on TS1 . Export button leads to export page 👍

**Review by ivanchimeno — DISMISSED** [review]
Tested on TS2 :+1:

File: src/main/webapp/app/exercises/quiz/manage/quiz-management.route.ts [hunk 1]
```diff
@@ -23,16 +23,16 @@ export const quizManagementRoute: Routes = [
         canDeactivate: [PendingChangesGuard],
     },
     {
-        path: ':courseId/quiz-exercises/:exerciseId/re-evaluate',
-        component: QuizReEvaluateComponent,
+        path: ':courseId/quiz-exercises/export',
+        component: QuizExerciseExportComponent,
         data: {
-            authorities: [Authority.INSTRUCTOR, Authority.ADMIN],
+            authorities: [Authority.TA, Authority.EDITOR, Authority.INSTRUCTOR, Authority.ADMIN],
```
Comment (niclasschuemann) [inline src/main/webapp/app/exercises/quiz/manage/quiz-management.route.ts:29]: The TA is not supposed to be able to export quiz exercises, so I guess we should remove the respective authority

File: src/main/webapp/app/exercises/quiz/manage/quiz-management.route.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export const quizManagementRoute: Routes = [
         redirectTo: ':courseId/exercises',
     },
     {
-        path: ':courseId/quiz-exercises/:exerciseId',
+        path: ':courseId/quiz-exercises/new',
         component: QuizExerciseDetailComponent,
         data: {
             authorities: [Authority.TA, Authority.EDITOR, Authority.INSTRUCTOR, Authority.ADMIN],
```
Comment (niclasschuemann) [inline src/main/webapp/app/exercises/quiz/manage/quiz-management.route.ts:19]: The TA is not supposed to access the quiz exercise creation view, so we should remove the respective authority

**Review by niclasschuemann — CHANGES_REQUESTED** [review]
The changes work as expected on TS1, but I have added a few comments regarding the authorities. If you think this is out of scope for this PR, feel free to dismiss this review 😄

File: src/main/webapp/app/exercises/quiz/manage/quiz-management.route.ts [hunk 3]
```diff
@@ -42,23 +42,23 @@ export const quizManagementRoute: Routes = [
...
         canActivate: [UserRouteAccessService],
-        canDeactivate: [PendingChangesGuard],
     },
     {
-        path: ':courseId/quiz-exercises/export',
-        component: QuizExerciseExportComponent,
+        path: ':courseId/quiz-exercises/:exerciseId/edit',
+        component: QuizExerciseDetailComponent,
         data: {
             authorities: [Authority.TA, Authority.EDITOR, Authority.INSTRUCTOR, Authority.ADMIN],
```
[the remark this answers is not in the document]
Comment (imstreetspirit) [reply src/main/webapp/app/exercises/quiz/manage/quiz-management.route.ts:57]: hmm, okay I did not check the other routes permissions. Thanks for capturing, I think I can handle this in this PR :)

**Review by argertboja — COMMENTED** [review]
Tested on TS2 and everything works fine. 👍🏼 

 Something I would suggest is to disable the export button in case there are no quizzes selected to export. See image below:
 
image

**Comment by imstreetspirit** [comment]
> Tested on TS2 and everything works fine. 👍🏼
> 
> Something I would suggest is to disable the export button in case there are no quizzes selected to export. See image below:
> 
> image

hmm, could be part of ui improvement, but I think that it is not relevant with the problem we have now. I just took noted this.

**Review by lschlesinger — APPROVED** [review]
Tested on TS3, worked as described 👍

**Review by real-dmy — APPROVED** [review]
Tested on TS1. Works as expected.

**Review by FrankeLukas — APPROVED** [review]

**Review by rs-maker — APPROVED** [review]
Works on local too :+1:

**Review by katjanakosic — APPROVED** [review]
Worked as expected on TS2

**Review by canyelten — APPROVED** [review]
Tested on TS1, works as expected.

**Pull request merged**